src/integrations/__tests__/unixIntegrationManager.spec.ts

```diff
@@ -0,0 +1,300 @@
+  if (testDir) {
+    await fs.promises.rm(testDir, { recursive: true, force: true });
+  }
+});
+
+test('Ensure symlinks and dirs are created properly', async() => {
+  const integrationManager = new UnixIntegrationManager(
+    resourcesDir, integrationDir, dockerCliPluginDir);
+
+  await integrationManager.enforce();
+  expect(fs.promises.readdir(integrationDir)).resolves.not.toThrow();
+  for (const name of await integrationManager.getIntegrationNames()) {
+    const integrationPath = path.join(integrationDir, name);
+
+    expect(fs.promises.readlink(integrationPath, 'utf8')).resolves.not.toThrow();
+  }
+  for (const name of await integrationManager.getDockerCliPluginNames()) {
+    const pluginPath = path.join(dockerCliPluginDir, name);
+
+    expect(fs.promises.readlink(pluginPath, 'utf8')).resolves.not.toThrow();
+  }
+});
+
+test('Ensure symlinks and dirs are removed properly', async() => {
+  await createTestSymlinks(resourcesDir, integrationDir, dockerCliPluginDir);
+  const integrationManager = new UnixIntegrationManager(
+    resourcesDir, integrationDir, dockerCliPluginDir);
+
+  await integrationManager.remove();
+  expect(fs.promises.readdir(integrationDir)).rejects.toThrow();
+  expect(fs.promises.readdir(dockerCliPluginDir)).resolves.toEqual([]);
+});
+
+test('Existing docker CLI plugins should not be overwritten upon .enforce()', async() => {
+  // create existing plugin
+  const existingPluginPath = path.join(dockerCliPluginDir, 'docker-compose');
+  const existingPluginContents = 'meaningless contents';
+
+  await fs.promises.mkdir(dockerCliPluginDir, { mode: 0o755 });
+  await fs.promises.writeFile(existingPluginPath, existingPluginContents);
+
+  const integrationManager = new UnixIntegrationManager(
+    resourcesDir, integrationDir, dockerCliPluginDir);
+
+  await integrationManager.enforce();
+
+  const newContents = await fs.promises.readFile(existingPluginPath, 'utf8');
+
+  expect(newContents).toEqual(existingPluginContents);
+});
+
+test('Existing docker CLI plugins should not be removed upon .remove()', async() => {
+  // create existing plugin
+  const existingPluginPath = path.join(dockerCliPluginDir, 'docker-compose');
+  const existingPluginContents = 'meaningless contents';
+
+  await fs.promises.mkdir(dockerCliPluginDir, { mode: 0o755 });
+  await fs.promises.writeFile(existingPluginPath, existingPluginContents);
+
+  const integrationManager = new UnixIntegrationManager(
+    resourcesDir, integrationDir, dockerCliPluginDir);
+
+  await integrationManager.remove();
+
+  const newContents = await fs.promises.readFile(existingPluginPath, 'utf8');
+
+  expect(newContents).toEqual(existingPluginContents);
+});
+
+test('.enforce() should be idempotent', async() => {
+  const integrationManager = new UnixIntegrationManager(
+    resourcesDir, integrationDir, dockerCliPluginDir);
+
+  await integrationManager.enforce();
+
+  return integrationManager.enforce();
```

Comment:
With "idempotent" in the test description, I would have expected code to enforce that nothing has changed after the second call to `enforce`. Like do a `readdir` after the first call, then again after the second, and verify the arrays are the same.